resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentCredentialServiceImpl.java

```diff
@@ -0,0 +1,275 @@
+				//		NotificationTemplateCode.RS_CREDENTIAL_FAILURE, null);
+				//throw new ResidentServiceException(ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorCode(),
+					//	ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorMessage());
+			}
+		}/* catch (OtpValidationFailedException e) {
+			sendNotification(dto.getIndividualId(), IdType.valueOf("UIN"),
+					NotificationTemplateCode.RS_CREDENTIAL_FAILURE, null);
+			throw new ResidentServiceException(ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorCode(),
+					e.getErrorText(), e);
+		}*/
+	/*	catch (ResidentServiceCheckedException e) {	
+			//sendNotification(dto.getIndividualId(), IdType.valueOf("UIN"),
+					//NotificationTemplateCode.RS_CREDENTIAL_FAILURE, null);
+			throw new ResidentServiceException(ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorCode(),
+					ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorMessage(), e);
+		}*/
+		catch (ApisResourceAccessException e) {
+		//	sendNotification(dto.getIndividualId(), IdType.valueOf("UIN"),
+			//		NotificationTemplateCode.RS_CREDENTIAL_FAILURE, null);
+			throw new ResidentServiceException(ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorCode(),
+					ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorMessage(), e);
+		}
+		catch (IOException e) {
+			//sendNotification(dto.getIndividualId(), IdType.valueOf("RID"),
+				//	NotificationTemplateCode.RS_CREDENTIAL_FAILURE, null);
+			throw new ResidentServiceException(ResidentErrorCode.IO_EXCEPTION.getErrorCode(),
+					ResidentErrorCode.IO_EXCEPTION.getErrorMessage(), e);
+		}
+
+		return residentCredentialResponseDto;
+	}
+	
+	@Override
+	public CredentialRequestStatusResponseDto getStatus(String requestId)
+			throws ResidentServiceCheckedException {
+		ResponseWrapper<CredentialRequestStatusResponseDto> responseDto = null;
+		CredentialRequestStatusResponseDto credentialRequestStatusResponseDto=new CredentialRequestStatusResponseDto();
+		try {
+			String credentialUrl=credentailStatusUrl+requestId;
+			URI credentailStatusUri = URI.create(credentialUrl);
+			responseDto =residentServiceRestClient.getApi(credentailStatusUri, ResponseWrapper.class, tokenGenerator.getToken());
+			credentialRequestStatusResponseDto = JsonUtil
+						.readValue(JsonUtil.writeValueAsString(responseDto.getResponse()), CredentialRequestStatusResponseDto.class);
+		} catch (ApisResourceAccessException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		catch(IOException e) {
+			e.printStackTrace();
+		}
+		return credentialRequestStatusResponseDto;
+	}
+
+	public CredentialReqestDto prepareCredentialRequest(ResidentCredentialRequestDto residentCreDto) {
+		CredentialReqestDto crDto=new CredentialReqestDto();
+		crDto.setAdditionalData(residentCreDto.getAdditionalData());
+		crDto.setCredentialType(residentCreDto.getCredentialType());
+		crDto.setEncrypt(residentCreDto.isEncrypt());
+		crDto.setEncryptionKey(residentCreDto.getEncryptionKey());
```

Comment:
IF user not provided then please generate encryption key using public String generatePin() {
	return  RandomStringUtils.randomAlphabetic(5);
	} 
give this encryption pin in notification template